blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/EnumMetric.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2013 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.types;
+
+import com.rackspacecloud.blueflood.exceptions.InvalidDataException;
+import com.rackspacecloud.blueflood.utils.TimeValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+
```

Comment:
@VinnyQ I think we can add a field in the BluefloodEnumRollup class like this. 

```
HashSet<String> stringEnumValues = new HashSet<String>() 
```

Then IMO, you will not need this EnumMetric class, you should just be able to use the PreaggregatedMetric class. 
Notice that PreaggregatedMetric already has a Rollup field. 
https://github.com/rackerlabs/blueflood/blob/10ca96311e71adc913ee30458ef70d83423399f5/blueflood-core/src/main/java/com/rackspacecloud/blueflood/types/PreaggregatedMetric.java#L28

We add a method to retrieve the rollup in PreaggregatedMetric class as follows. 

```
 public Rollup getRollup() {
        return this.value;
 }
```

Then you can modify your EnumElasticIO in this way: 

```
if (obj instanceof PreaggregatedMetric ) {
                PreaggregatedMetric metric = (PreaggregatedMetric)obj;
                if (metric.getRollupType() == RollupType.ENUM) {
                    BluefloodEnumRollup enumRollup = (BluefloodEnumRollup)metric.getRollup();
                    Locator locator = metric.getLocator();
                    Discovery md = new Discovery(locator.getTenantId(), locator.getMetricName());

                    Map<String, Object> info = new HashMap<String, Object>();

                    HashSet<String> enumValues = enumRollup.getEnumValues();

                    if (enumValues != null) {
                        info.put(ESFieldLabel.enum_values.toString(), enumValues);
                    }
                }
            }
```

It makes sense to use the Rollup class to store values, because we are going to build them there anyways right? With the bad metrics table. 

I like this idea of EnumMetric and having separate Metric classes. Unfortunately, currently the metric type specific logic is only in Rollups.